src/nagini_translation/translators/permission.py

```diff
@@ -36,14 +38,113 @@ def translate_perm_Num(self, node: ast.Num, ctx: Context) -> Expr:
         raise UnsupportedException(node)
 
     def translate_perm_BinOp(self, node: ast.BinOp, ctx: Context) -> Expr:
+        def translate_node(node):
```

Comment:
We've tried to avoid nested functions pretty much everywhere; could you move this outside?